### ISSUE TITLE
numpy() and from_numpy() for HalfTensor

### ISSUE THREAD

**Issue by kohr-h** [pytorch/pytorch#2943]
`HalfTensor` has no to/from Numpy array methods. As written by @apaszke in #2935 that would be possible.

**Comment by kohr-h** [comment]
With some guidance and if it makes sense I can do it. The first attempt doing the obvious addition of 
```C
#ifdef TH_REAL_IS_HALF
#define NUMPY_TYPE_ENUM NPY_HALF
#endif
```
in `torch/csrc/generic/Tensor.cpp` gives compile errors
```
[...]/torch/csrc/generic/Tensor.cpp:34:13: error: no match for ‘operator=’ (operand types are ‘THHalf {aka __THHalf}’ and ‘double’)
     data[i] = arrdata[i];                                 \
             ^
```
and similar for other types.

**Comment by apaszke** [comment]
Can you paste a complete compile log into a GitHub gist and post the link in here?

**Comment by kohr-h** [comment]
Here we go: [compile.log](https://gist.github.com/kohr-h/7278540929c6cdcf123d12f1eb6e2831)

**Comment by apaszke** [comment]
Just change [this part](https://github.com/pytorch/pytorch/blob/master/torch/csrc/generic/Tensor.cpp#L29-L32) to a single call to `memcpy`

**Comment by kohr-h** [comment]
Closed by #2953
